routing/unified_policies.go

```diff
@@ -195,19 +201,22 @@ func (u *unifiedPolicy) getPolicyLocal(amt lnwire.MilliSatoshi,
 		bestPolicy = edge.policy
 	}
 
-	return bestPolicy
+	// We return a zero capacity here to not apply a capacity factor cutoff
+	// for local channels, because we know that the amount should succeed.
```

Comment:
This sounds like a layering violation. Perhaps clear the capacity higher up?

Or is this an indication that `getPolicy` isn't the right method anymore because it sometimes does and sometimes does not return a capacity?

I know this caused by merging capacity determination into `getPolicy`, but maybe there is a better structure to do a single iteration without dead return values.

Reply:
What could be useful too is this commit: https://github.com/lightningnetwork/lnd/pull/6703/commits/b1bbf3fc42aa60bf6baa174a027394c81a83654e

It allows some extra data to be returned instead of just the db type `CachedEdgePolicy`. I thought that makes sense as we are constructing a sort of synthetic edge that blends many real edges together. Over time, this may evolve towards not exposing the raw `CachedEdgePolicy` at all anymore.

Reply:
I followed your suggestion and returned a `unifiedPolicyEdge`, although I applied only the minimal change compared to your commit. I think that this really changes the concept of unified policies to unified edges. 

Perhaps we could also do a refactor: `unifiedPolicies` -> `nodeEdgeUnifier`, `unifiedPolicy` -> `edgeUnifier`, and `unifiedPolicyEdge` -> `unifiedEdge`. Also replacing all references to a `policy` with `edge` or `unifiedEdge`.

Reply:
Yes, those renames make sense. I was confused again by the naming when I first looked at this again.

Reply:
ok, I refactored that, hope it makes more sense now and I didn't change too much